multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -416,9 +405,15 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
 	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+
+	switch c.clusterRole {
+	case LeaderCluster:
+		go wait.UntilWithContext(ctx, c.cleanupMemberClusterAnnounces, 12*time.Hour)
+	case MemberCluster:
+		if err := c.RunOnce(ctx); err != nil {
+			c.Enqueue()
+			go wait.UntilWithContext(ctx, c.runWorker, time.Second)
```

Comment:
How one second is chosen here?

Reply:
Good catch, the original value is one second. I kept the old value. I think there will be a queue rate limit to control re-queue to truly run the clean up functions. But it's indeed unnecessary to run so frequently. I will update it to 5 seconds.